cmd/nardump/nardump_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package main
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+// SubCmdFlags space separated list of command line flags.
+const SubCmdFlags = "SUB_CMD_FLAGS"
+
+// setupTmpdir sets up a known golden layout, covering all allowed file/folder types in a nar
+func setupTmpdir(t *testing.T) string {
+	tmpdir := t.TempDir()
+	pwd, _ := os.Getwd()
+	os.Chdir(tmpdir)
+	defer os.Chdir(pwd)
+	os.MkdirAll("sub/dir", 0755)
+	os.Symlink("brokenfile", "brokenlink")
+	os.Symlink("sub/dir", "dirl")
+	os.Create("sub/dir/file1")
+	f, _ := os.Create("file2m")
+	_ = f.Truncate(2 * 1024 * 1024)
+	f.Close()
+	os.Symlink("../file2m", "sub/goodlink")
+	return tmpdir
+}
```

Comment:
I'd prefer we avoid using a subprocess of ourselves. Those tests (while often necessary) are such a pain in the butt to maintain and keep portable that I try to avoid them when possible. And it seems totally possible here... just test `writeNAR` instead of `main`.

Reply:
Alright, I will change the testing to test just writeNar.
I have never done it before, so I found it to be very interesting. I didn't think it was bad for long term mantainance.

Reply:
Done.